### PR TITLE
Transparency oversight

### PR DESCRIPTION
**Pull request by MikeMcGarry**

### CODE REVIEW

File: lusid-sdk-python/tests/finbournetest.py [hunk 1]
```diff
@@ -431,6 +442,154 @@ def create_property_definition_test(self, property_request, property):
...
+
+        # Convert our reconciliation breaks to a dictionary with the instrument uid as the key
+        rec_breaks = {rec_break.instrument_uid: rec_break for rec_break in reconciliation_break_response.values}
+
+        # Iterate over our left holdings and check that any breaks are correct
+        for key, left_holding in left_holdings.items():
+            # If there is more than one holding for an instrument, aggregate the holdings
+            left_holding_units, left_holding_cost_amount, left_holding_cost_currency = aggregate_holdings(left_holding)
+            # Try and find a holding on the right side for the same instrument
+            try:
```
Comment (rizwansaeed) [inline lusid-sdk-python/tests/finbournetest.py]: can this be replaced with an ` if ... in ...` rather than using exceptions for control logic

File: lusid-sdk-python/tests/finbournetest.py [hunk 2]
```diff
@@ -431,6 +442,154 @@ def create_property_definition_test(self, property_request, property):
...
+                                                   filter="transactionId eq '{}'".format(transaction_id)).values[0]
+
+        # Create a dictionary with the key, value pairs for each of our properties on the transaction
+        transaction_properties = {trans_property.key:trans_property.value for trans_property in transaction.properties}
+
+        # Check that the property exists and that the value is correct
+        self.assertTrue(property_key in transaction_properties)
+        self.assertEqual(property_value, transaction_properties[property_key])
+
     def test(self):
```
Comment (rizwansaeed) [inline lusid-sdk-python/tests/finbournetest.py]: why is this needed, can it be removed?

File: lusid-sdk-python/tests/finbournetest.py [hunk 3]
```diff
@@ -431,6 +442,154 @@ def create_property_definition_test(self, property_request, property):
         self.assertEqual(property.data_type_id.scope, property.data_type_id.scope)
         self.assertEqual(property.data_type_id.code, property.data_type_id.code)
 
+    def reconciliation_asserts(self,
```
Comment (rizwansaeed) [inline lusid-sdk-python/tests/finbournetest.py]: why does this not use `/api/portfolios/$reconcileholdings`?

File: lusid-sdk-python/tests/finbournetest.py (hunk 2 above)
Comment (MikeMcGarry) [reply lusid-sdk-python/tests/finbournetest.py]: Yes it can be removed. I thought it was necessary for things to work correctly, but apparently not. I'll double check using Travis removing this has not broken anything.

File: lusid-sdk-python/tests/finbournetest.py (hunk 1 above)
Comment (MikeMcGarry) [reply lusid-sdk-python/tests/finbournetest.py]: Done

File: lusid-sdk-python/tests/finbournetest.py (hunk 3 above)
Comment (MikeMcGarry) [reply lusid-sdk-python/tests/finbournetest.py]: As discussed, these assertions are used to verify that the reconciliation has returned the correct result. Now to check that the reconciliation has returned what we expected we need to work out the difference between our holdings in the internal scope and those in the fund accountant scope and check that this is what the reconciliation returned as breaks. 

This is somewhat challenging as it requires me to compare holdings which I don't have the raw data for. This is because I have the raw data for the initial holdings and I have the raw data for the transactions, but adding the transactions to the initial holdings is done inside LUSID and thus I haven't stored locally the updated holdings which we need to compare.

This is why I have used get_holdings to get the holdings from LUSID and then compared them with the returned reconciliation breaks rather than passing in the raw result that I expect.

I think that this is a better approach then trying to manually add my transactions to my holdings outside of LUSID as we need to check that the reconciliation has worked at multiple different effectiveAt times.

What do you think?

Comment (rizwansaeed) [reply lusid-sdk-python/tests/finbournetest.py]: We should be able to do something like https://github.com/finbourne/lusid-sdk-csharp/blob/master/lusid-sdk-csharp/LusidSdk.Tests/LusidApiTests.cs#L483, we shouldn't be testing that the functionality itself works as that is tested inside LUSID

Comment (MikeMcGarry) [reply lusid-sdk-python/tests/finbournetest.py]: As per our discussion abstracted away
